src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py

```diff
@@ -300,11 +300,15 @@ def extrapolate(self):
         self._update_calculator()
         try:
             params, extrapolation, _ = self._calculator.compute_extrapolation()
+            self.model.itemChanged.disconnect(self.model_changed)
+
             self.model.setItem(W.W_GUINIERA, QtGui.QStandardItem("{:.3g}".format(params['A'])))
             self.model.setItem(W.W_GUINIERB, QtGui.QStandardItem("{:.3g}".format(params['B'])))
             self.model.setItem(W.W_PORODK, QtGui.QStandardItem("{:.3g}".format(params['K'])))
             self.model.setItem(W.W_PORODSIGMA,
                                QtGui.QStandardItem("{:.4g}".format(params['sigma'])))
+            self.model.itemChanged.connect(self.model_changed)
+            self.model_changed(None)
```

Comment:
`model_changed()` calls `draw_q_space()`, which clears the figure and draws.  Two lines below there is another call to `draw_q_space()` which again clears and draws.

The try/except only needs to cover `compute_extrapolation()`. 

Should the A, B, K and sigma parameters on the form be cleared if the calculation fails?

Reply:
Good catch